base_tier_validation/models/tier_validation.py

```diff
@@ -472,16 +482,11 @@ def request_validation(self):
                     for td in tier_definitions:
                         if rec.evaluate_tier(td):
                             sequence += 1
-                            created_trs += tr_obj.create(
-                                {
-                                    "model": self._name,
-                                    "res_id": rec.id,
-                                    "definition_id": td.id,
-                                    "sequence": sequence,
-                                    "requested_by": self.env.uid,
-                                }
+                            vals_list.append(
+                                rec._prepare_tier_review_vals(td, sequence)
                             )
                     self._update_counter({"review_created": True})
+        created_trs += tr_obj.create(vals_list)
```

Comment:
No need now of pre-setting the variable:

```suggestion
        created_trs = tr_obj.create(vals_list)
```

(and remove its initialization).

And as said, you can also reduce one indentation level joining the 2 if.